mavproxy/source/docs/development/mavdevenvlinux.rst

```diff
@@ -13,20 +13,43 @@ The pip-installed MAVProxy will need to uninstalled (if already installed) to pr
 
 .. code:: bash
 
-    pip uninstall MAVProxy
+    python3 -m pip uninstall MAVProxy
 
 Use git to download the MAVProxy source:
 
 .. code:: bash
 
     git clone https://github.com/ArduPilot/MAVProxy.git
 
-After making the desired changes, MAVProxy is required to be installed
-(any change to the modules won't work otherwise). This needs to happen after any
-changes to the source code. This can be done by:
 
-.. code:: bash
 
-    python setup.py build install --user
+.. tabs::
+
+   .. group-tab:: User Python Environment
+
+    After making the desired changes, MAVProxy is required to be installed
+    (any change to the modules won't work otherwise). This needs to happen after any
+    changes to the source code. This can be done by:
+
+      .. code-block:: bash
+
+        python3 setup.py build install --user
+
+
+   .. group-tab:: Virtual Python Environment
+
+    Using a virtual environment allows quicker editing.
+    If you make code changes, just re-run MAVProxy.
+
+      .. code-block:: bash
+
+        python3 -m venv .venv
+        source .venv/bin/activate
+        python3 setup.py develop

```

Comment:
My understanding is that running `setup.py` directly is very much deprecated.

Reply:
Yep, https://packaging.python.org/en/latest/discussions/setup-py-deprecated/#what-commands-should-be-used-instead